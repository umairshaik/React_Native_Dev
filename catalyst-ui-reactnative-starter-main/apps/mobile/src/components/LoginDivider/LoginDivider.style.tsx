import {View} from 'react-native';
import styled from 'styled-components/native';
import Divider from '../Divider/Divider';

export const BoxView = styled(View)`
  border-color: #d3d3d3;
  border-width: 1px;
  height: 21px;
  width: 39px;
  border-radius: 4px;
  justify-content: center;
`;
export const DividerL = styled(Divider)`
  width: 7px;
  border-width: 2px;
  align-self: center;
`;
