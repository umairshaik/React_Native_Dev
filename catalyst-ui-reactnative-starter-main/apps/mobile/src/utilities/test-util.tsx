import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {ThemeProvider} from '../context/theme/themeContext';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});
const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <ThemeProvider>
    <QueryClientProvider client={query}>{children}</QueryClientProvider>
  </ThemeProvider>
);

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export * from '@testing-library/react-native';
export {customRender as render};
