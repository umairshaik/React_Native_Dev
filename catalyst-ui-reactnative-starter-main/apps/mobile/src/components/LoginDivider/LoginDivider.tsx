import {DividerL, BoxView} from './LoginDivider.style';

const Box = () => (
  <BoxView>
    <DividerL accessibilityProps="Box Divider" />
  </BoxView>
);

export default Box;
