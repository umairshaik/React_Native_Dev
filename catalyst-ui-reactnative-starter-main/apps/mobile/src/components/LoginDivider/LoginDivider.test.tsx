import {render} from '../../utilities/test-util';
import LoginDivider from './LoginDivider';

describe('[component]-[LoginDivider]', () => {
  test('should render the LoginDivider component', () => {
    const Box = render(<LoginDivider />);
    expect(Box).toMatchSnapshot();
  });
});
