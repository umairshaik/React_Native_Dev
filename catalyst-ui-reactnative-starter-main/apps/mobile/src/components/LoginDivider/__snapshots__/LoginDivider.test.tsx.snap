// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component]-[LoginDivider] should render the LoginDivider component 1`] = `
<View
  style={
    [
      {
        "borderColor": "#d3d3d3",
        "borderRadius": 4,
        "borderWidth": 1,
        "height": 21,
        "justifyContent": "center",
        "width": 39,
      },
    ]
  }
>
  <View
    accessibilityHint="Box Divider"
    accessibilityLabel="click to Box Divider"
    style={
      [
        {
          "borderBottomColor": "#000",
          "borderBottomWidth": 0.5,
          "height": 0,
        },
        {
          "alignSelf": "center",
          "borderWidth": 2,
          "width": 7,
        },
      ]
    }
  />
</View>
`;
